src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -345,8 +347,16 @@ CitusBeginModifyScan(CustomScanState *node, EState *estate, int eflags)
 
 	if (ModifyJobNeedsEvaluation(workerJob))
 	{
-		/* evaluate both functions and parameters */
-		ExecuteMasterEvaluableFunctionsAndParameters(jobQuery, planState);
+		if (jobQuery->commandType == CMD_SELECT)
+		{
+			/* only functions inside modifying CTEs will be evaluated */
+			ExecuteMasterEvaluableParameters(jobQuery, planState);
```

Comment:
Where/how does the function evaluation inside modifying CTEs happen?

Reply:
I updated the evaluation logic to switch to function evaluation when it recurses into a query which isn't `CMD_SELECT`

Reply:
Could we move this `if` statement into the ExecuteMasterEvaluable... function in that case?